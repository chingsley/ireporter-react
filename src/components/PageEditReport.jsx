import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import FormReport from './FormReport';

class EditReportPage extends Component {
  constructor(props) {
    super(props);
    this.history = props.history;
  }

  render() {
    // eslint-disable-next-line react/prop-types
    const { userReports, match: { params: { id } } } = this.props;
    const report = userReports.filter(singleReport => singleReport.id === Number(id));

    return (
      <div className="report-container">
        <h2 className="report-container__header">EDIT REPORT</h2>
        <FormReport {...this.props} report={report[0]} mode="edit" />
      </div>
    );
  }
}

EditReportPage.propTypes = {
  history: PropTypes.oneOfType([PropTypes.object, PropTypes.func]).isRequired,
  userReports: PropTypes.instanceOf(Array).isRequired,
};

const mapStateToProps = state => ({
  userReports: state.report.userReports,
});

export default connect(mapStateToProps, {})(EditReportPage);
