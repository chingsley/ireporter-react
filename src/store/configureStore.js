import {
  applyMiddleware, createStore, combineReducers, compose,
} from 'redux';
import thunk from 'redux-thunk';
import authReducer from '../reducers/authReducer';

export default () => {
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // eslint-disable-line
  const store = createStore(
    combineReducers({
      auth: authReducer,
    }),
    composeEnhancers(applyMiddleware(thunk)),
  );
  return store;
};
