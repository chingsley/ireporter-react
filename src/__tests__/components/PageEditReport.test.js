import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import PageEditReport from '../../components/PageEditReport';

const wrap = () => {
  const props = {
    history: {
      push: jest.fn(),
      location: {
        pathname: '',
      },
    },
    match: {
      params: {
        id: 1,
      },
    },

    authorise: jest.fn(),
    isLoggedIn: false,
    fetching: false,
    formType: 'signup',
  };

  const mockStore = configureStore([thunk]);

  const store = mockStore({
    auth: {
      isLoggedIn: true,
      user: {},
    },
    fetchStatus: {
      fetching: false,
      errMessage: '',
    },
    report: {
      report: {},
      userReports: [
        {
          id: 1,
          comment: 'some comments here',
          location: '1.2, 2.1',
          status: 'draft',
          createdOn: Date.now(),
          Images: ['a-given-image.png'],
        },
        {
          id: 2,
          comment: 'some comments here',
          location: '1.2, 2.1',
          status: 'draft',
          createdOn: Date.now(),
          Images: ['another-img.npn'],
        },
      ],
    },
  });

  const mountWrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <PageEditReport {...props} />
      </BrowserRouter>
    </Provider>,
  );
  return { mountWrapper };
};

describe('Test the report form', () => {
  const { shallowWrapper: wrapper } = wrap();
  it('should render with the correct elements', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
