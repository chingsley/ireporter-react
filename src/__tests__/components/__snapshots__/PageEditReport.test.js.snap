// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the report form should render with the correct elements 1`] = `undefined`;
